File: src/main/java/bernard/tatin/ProcFS/ProcessCommandLine.java
package bernard.tatin.ProcFS;

import bernard.tatin.Tools.ForFiles;

public class ProcessCommandLine {
    final private static ProcessCommandLine ourInstance = new ProcessCommandLine();
    private static String commandLine = null;

    public static String getCommandLine() {
        if (commandLine == null) {
            try {
                commandLine = ForFiles.loadTextFile(LinuxProc.procPathName("cmdline") );
                if (commandLine == null) {
                    commandLine = "<no command line found>";
                }
            }
            catch (Exception e) {
                commandLine = "<no command line found>";
            }
        }
        return commandLine;
    }

    private ProcessCommandLine() {
    }
}

File: kotlin-version/src/main/kotlin/bernard/tatin/ProcFS/ProcessCommandLine.kt
package bernard.tatin.ProcFS

import bernard.tatin.Tools.ForFiles

object ProcessCommandLine {
    //    final private static ProcessCommandLine ourInstance = new ProcessCommandLine();
    private var commandLine: String? = null

    fun getCommandLine(): String {
        if (commandLine == null) {
            try {
                commandLine = ForFiles.loadTextFile(LinuxProc.procPathName("cmdline"))
                if (commandLine == null) {
                    commandLine = "<no command line found>"
                }
            } catch (e: Exception) {
                commandLine = "<no command line found>"
            }

        }
        return commandLine
    }
}
